### PR TITLE
chore: Change usage of ChooseOne (no final condition)

### PR DESCRIPTION
**Pull request by presleyp**
Based on our conversation in Frontend Variety, I made the `condition` prop optional in `Cond` component. `ChooseOne` will throw an error if you use one on your default case, or don't use one in a non-final case.

### CODE REVIEW

File: site/src/components/Conditionals/ChooseOne.tsx [hunk 1]
```diff
@@ -21,9 +20,20 @@ export const Cond = ({ children, condition }: PropsWithChildren<CondProps>): JSX
...
  * @returns one of its children
+ * @throws an error if its last child has a condition prop, or any non-final children do not have a condition prop
  */
 export const ChooseOne = ({ children }: PropsWithChildren): JSX.Element => {
   const childArray = Children.toArray(children) as JSX.Element[]
-  const chosen = childArray.find((child) => child.props.condition)
-  return chosen ?? childArray[childArray.length - 1]
+  const conditionedOptions = childArray.slice(0, childArray.length - 1)
+  const defaultCase = childArray[childArray.length - 1]
+  if (defaultCase.props.condition !== undefined) {
```
Comment (code-asher) [inline site/src/components/Conditionals/ChooseOne.tsx:29]: idk if this is worth consideration but if given an empty array `defaultCase` would be `undefined` I think?

**Review by code-asher — APPROVED** [review]
Makes sense!!  IMO this is even better than what we talked about in FEV.  :+1:

File: site/src/components/Conditionals/ChooseOne.tsx (hunk 1 above)
Comment (presleyp) [reply site/src/components/Conditionals/ChooseOne.tsx:32]: Thanks! How does this look?

Comment (code-asher) [reply site/src/components/Conditionals/ChooseOne.tsx:32]: Perfect!!

**Pull request merged**